src/fairseq2/utils/warn.py

```diff
@@ -23,3 +23,11 @@ def enable_deprecation_warnings() -> None:
 
 def _warn_deprecated(msg: str) -> None:
     warnings.warn(msg, DeprecationWarning, stacklevel=2)
+
+
+# TODO: Remove in v0.13
```

Comment:
Can we create a test or something in the CI to track what needs to be removed at a particular dev version before we release (not part of this PR ofc)? I imagine this will become an issue very soon to keep track of all of the deprecations.

Reply:
How about extending `_warn_deprecated` with a mandatory `version` parameter? In `_warn_deprecated` we can then throw an `InternalError` if the current version of fairseq2 matches the specified `version`. Something like:

```python
from packaging.Version

def _warn_deprecated(msg: str, eol_version: Version) -> None:
  if fairseq2.__version__ >= eol_version:
    raise InternalError("Deprecated feature not removed at or after end-of-life version")
```

Reply:
This is a neat solution, the CI will catch that while building for the target version at worst!

Reply:
https://github.com/orgs/fairinternal/projects/173/views/1?pane=issue&itemId=134623353&issue=facebookresearch%7Cfairseq2%7C1387